### 1. train.csv 불러오기
### 2. 상위 5개 행 출력

In [17]:
import pandas as pd

df =pd.read_csv("train.csv")

df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


### 3. 열 정보 확인


In [18]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


**[분석] 결측치가 존재하는 열과 결측 개수**

전체 데이터는 891행이며, Non-Null Count가 891보다 작은 열이 결측치가 있는 열이다.

| 열 | 결측 개수 |
| --- | --- |
| Age | 177개 |
| Cabin | 687개 |
| Embarked | 2개 |

**[분석] dtype이 예상과 다르거나 주의가 필요한 열**

| 열 | dtype | 주의할 점 |
| --- | --- | --- |
| Age | float64 | 나이는 정수지만, 결측치(NaN)가 float 타입이라 열 전체가 float64로 저장됨 |
| Survived, Pclass | int64 | 숫자지만 크기의 의미가 없는 범주형 값(생존 여부, 객실 등급)임 |
| Sex | str | 문자열이라 분석·계산에 쓰려면 숫자로 바꿔야 함 (12번에서 인코딩) |

### 4. Age, Fare의 평균·최솟값·최댓값

In [19]:
import pandas as pd
df = pd.read_csv("train.csv")

print("Age 평균:", df["Age"].mean())
print("Age 최댓값:", df["Age"].max())
print("Age 최솟값:", df["Age"].min())

print("Fare 평균:", df["Fare"].mean())
print("Fare 최댓값:", df["Fare"].max())
print("Fare 최솟값:", df["Fare"].min())


Age 평균: 29.69911764705882
Age 최댓값: 80.0
Age 최솟값: 0.42
Fare 평균: 32.204207968574636
Fare 최댓값: 512.3292
Fare 최솟값: 0.0


### 5. 생존자·사망자 수

In [20]:
import pandas as pd
df =pd.read_csv("train.csv")

counts = df["Survived"].value_counts()

print(f"사망자(0): {counts[0]}명")
print(f"생존자(1): {counts[1]}명")

사망자(0): 549명
생존자(1): 342명


### 6. 객실 등급별 탑승객 수

In [21]:
counts = df["Pclass"].value_counts()
print(f"1등급: {counts[1]}명")
print(f"2등급: {counts[2]}명")
print(f"3등급: {counts[3]}명")

1등급: 216명
2등급: 184명
3등급: 491명


### 7. 50세 이상 탑승객 추출

In [22]:
df_50 = df[df["Age"]>= 50] 

df_50.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
6,7,0,1,"McCarthy, Mr. Timothy J",male,54.0,0,0,17463,51.8625,E46,S
11,12,1,1,"Bonnell, Miss. Elizabeth",female,58.0,0,0,113783,26.5500,C103,S
15,16,1,2,"Hewlett, Mrs. (Mary D Kingcome)",female,55.0,0,0,248706,16.0000,NaN,S
33,34,0,2,"Wheadon, Mr. Edward H",male,66.0,0,0,C.A. 24579,10.5000,NaN,S
54,55,0,1,"Ostby, Mr. Engelhart Cornelius",male,65.0,0,1,113509,61.9792,B30,C


### 8. 나이대(AgeGroup) 열 추가

In [23]:
def age_group(age):
    if pd.isna(age):
        return "미확인"
    elif age < 10:
        return "아동"
    elif age < 20:
        return "10대"
    elif age < 30:
        return "20대"
    elif age < 40:
        return "30대"
    elif age < 50:
        return "40대"
    elif age < 60:
        return "50대"
    else:
        return "60대 이상"

df["AgeGroup"]= df["Age"].apply(age_group)    

df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked,AgeGroup
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S,20대
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C,30대
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S,20대
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S,30대
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S,30대


### 9. 성별·객실 등급별 평균 생존율

In [24]:
df.groupby(["Sex","Pclass"])["Survived"].mean()

Sex     Pclass
female  1         0.968085
        2         0.921053
        3         0.500000
male    1         0.368852
        2         0.157407
        3         0.135447
Name: Survived, dtype: float64

### 10. 나이대(AgeGroup)별 평균 생존율